### PR TITLE
Check engine image state in toleration setting test

### PR DESCRIPTION
**Pull request by shuo-wu**
https://github.com/longhorn/longhorn/issues/734

### CODE REVIEW

File: manager/integration/tests/test_toleration_setting.py [hunk 1]
```diff
@@ -127,6 +127,14 @@ def wait_for_toleration_update(core_api, apps_api, count, set_tolerations):  # N
             time.sleep(RETRY_INTERVAL_LONG)
             continue
 
+        client = get_longhorn_api_client()
+        images = client.list_engine_image()
+        assert len(images) == 1
+        if images[0]["state"] != "ready":
+            updated = False
+            time.sleep(RETRY_INTERVAL_LONG)
+            continue
```
Comment (yasker) [inline manager/integration/tests/test_toleration_setting.py]: ??

Comment (yasker) [reply manager/integration/tests/test_toleration_setting.py]: It's confusing. Why add `time.sleep(RETRY_INTERVAL_LONG)` before every `continue` command? Is there a better way?

Comment (yasker) [reply manager/integration/tests/test_toleration_setting.py]: And there is a `continue` seems misaligned on line 116

Comment (shuo-wu) [reply manager/integration/tests/test_toleration_setting.py]: Puting `time.sleep(RETRY_INTERVAL_LONG)` at the top of the loop can avoid the case. But it looks weird.

Comment (yasker) [reply manager/integration/tests/test_toleration_setting.py]: I think it's OK to put it on the top. One extra second is not a big deal in the test.

**Pull request merged**